pyfr/solvers/base/elements.py

```diff
@@ -152,6 +152,14 @@ def set_backend(self, backend, nscal_upts):
         self._scal_upts_temp = backend.matrix(inb.ioshape, aliases=aliases,
                                               tags=inb.tags)
 
+        # Allocate required scratch space for artificial viscosity
+        if self.cfg.getfloat('solver-avis', 'amu0', 0.0):
+            self._avis_upts = avis = backend.matrix((nupts, 1, neles), tags={'align'})
+            self._avis_fpts = alloc('avis_fpts', (nfpts, 1, neles))
+            aliases = next((m for m in abufs if m.nbytes >= avis.nbytes), None)
```

Comment:
You probably don't want to be doing this.  Temporary buffers allocated like this can only be used outside of the flux evaluation code (e.g, filters).  Find another buffer and alias it explicitly.